### PR TITLE
DDPB-3911: Create elevated admin role

### PR DESCRIPTION
**Pull request by acsauk**
## Purpose
In preparation for giving case managers/admin users more responsibilities, we need something between the existing admin and super admin rights. This change creates an `ELEVATED_ADMIN_ROLE` that has the power to edit and add all user roles except for super admins. This is essentially the same as the existing admin role but by splitting this out we can assign certain actions to be limited to super and elevated admins only.

Fixes DDPB-3911

## Approach
I'v rejigged some of the UserVoter code to take into account the similarities for adding and editing a user. I also expanded the tests to cover all roles and ensured that PAs cant edit Profs (and vice versa) as they were previously allowed to do.

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have updated documentation (Confluence/ADR/tech debt doc) where relevant
- [x] I have added tests to prove my work
- [ ] The product team have approved these changes

### Frontend
- [ ] I have run an in-browser accessibility test (e.g. WAVE, Lighthouse)
- [ ] There are no deprecated CSS classes noted in the profiler
- [ ] Translations are used and the profiler doesn't identify any missing
- [ ] Any links or buttons added are screen reader friendly and contextually complete
- [ ] If adding GA events, I have updated or [checked](docs/runbooks/GOOGLE-ANALYTICS.md) the existing category or label values

### CODE REVIEW

File: api/config/security.yml [hunk 1]
```diff
@@ -130,7 +130,7 @@ security:
 
     # https://opgtransform.atlassian.net/wiki/spaces/DEPDS/pages/191299647/User+roles
```
Comment (Gugandeep) [inline api/config/security.yml:131]: Does the page in the wiki require updating?

File: client/templates/Report/Debt/edit.html.twig [hunk 2]
```diff
@@ -23,6 +23,7 @@
     {% for debt in form.debts %}
         <div class="{{ debt.moreDetails is defined ? ' js-details-expander ' : '' }}">
             {{ form_row(debt.debtTypeId) }}
+
```
Comment (Gugandeep) [inline client/templates/Report/Debt/edit.html.twig:26]: Was this intentional?

**Review by Gugandeep — DISMISSED** [review]
Good work. Looks good to go.

File: api/config/security.yml (hunk 1 above)
Comment (acsauk) [reply api/config/security.yml:131]: Very good point (and I'd never seen this page before 😂 ) I've updated it now

File: client/templates/Report/Debt/edit.html.twig (hunk 2 above)
Comment (acsauk) [reply client/templates/Report/Debt/edit.html.twig:26]: I had not - removed now 👍

File: client/src/Security/ClientVoter.php [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        $user = $token->getUser();
+
+        if (!$user instanceof User) {
+            // the user must be logged in; if not, deny access
+            return false;
+        }
+
+
+        switch ($attribute) {
+            case self::VIEW:
```
Comment (coderste) [inline client/src/Security/ClientVoter.php:61]: Do we need this case if it's not doing anything?

**Review by coderste — COMMENTED** [review]
Just left a question about one of the switch statements other than that this looks good to me 👍🏻

**Review by coderste — APPROVED** [review]

File: client/src/Security/ClientVoter.php (hunk 3 above)
Comment (acsauk) [reply client/src/Security/ClientVoter.php:61]: Stacking the cases above each other like this is just saying that if the attribute is view or edit then return the result of `$this->canManage()`

**Pull request merged**